packages/roosterjs-content-model-core/lib/corePlugin/selection/SelectionPlugin.ts

```diff
@@ -525,7 +497,11 @@ class SelectionPlugin implements PluginWithState<SelectionPluginState> {
 
     private selectBeforeOrAfterElement(editor: IEditor, element: HTMLElement, after?: boolean) {
         const doc = editor.getDocument();
-        const parent = element.parentNode;
+        let parent = element.parentNode;
+        if (isNodeOfType(parent, 'ELEMENT_NODE') && parent.shadowRoot && parent.parentNode) {
+            element = parent;
+            parent = parent.parentNode;
+        }
```

Comment:
why need this?

Reply:
when type over an image, the cursor was going into the shadow dom, to ensure the cursor is being placed out of the shadow dom, I added this code. it makes copy/cut possible.

Reply:
Hmm in my test branch, I'm selecting the whole image (or whole wrapper SPAN), then we don't need to do any special handling for keyboard event since the image is really selected. Can we still do that?

Or at least, can we handle shadow dom in ImageEdit plugin since SelectionPlugin should not know the detail about shadow DOM